### PR TITLE
Bump contributor requirement to NodeJS v16+

### PR DESCRIPTION
**Pull request by queengooborg**
This PR bumps the contributors' minimum NodeJS requirement to v16.  Based upon a poll in #12792, most of the top (and active) BCD contributors are not only 
fine with switching, but state that they only keep v14 around because of BCD.  Additionally, this bumps all workflows to use v16 to maintain consistency.

### CODE REVIEW

File: .github/workflows/add-push-artifacts.yml [hunk 1]
```diff
@@ -8,20 +8,20 @@ jobs:
...
-    - run: node ./scripts/diff-features.js --no-github --format=json > features.diff.json
-    - uses: actions/upload-artifact@v2
-      with:
-        name: enumerate-features
-        path: features.json
-    - uses: actions/upload-artifact@v2
-      with:
-        name: diff-features
-        path: features.diff.json
+      - uses: actions/checkout@v2
```
Comment (foolip) [inline .github/workflows/add-push-artifacts.yml:11]: Please undo the indentation changes, to limit this to just the version change. Indentation is currently inconsistent in our YAML files, which I think is fine. If we wanted to make it consistent let's use prettier or something that can change the indentation to whatever we want.

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -5,20 +5,20 @@ name: Node.js CI
 
 on:
   push:
-    branches: [ ] # Test on push to every branch
+    branches: [] # Test on push to every branch
```
Comment (foolip) [inline .github/workflows/node.js.yml:8]: Unrelated formatting changes.

**Review by foolip — CHANGES_REQUESTED** [review]

File: .github/workflows/add-push-artifacts.yml (hunk 1 above)
Comment (queengooborg) [reply .github/workflows/add-push-artifacts.yml:11]: Ah sorry, my editor's set to auto-format with Prettier on file save.  Let me see if I can tweak the files without formatting...

**Comment by queengooborg** [comment]
This was discussed during the last BCD meeting with a resounding "yes" from all of the owners, and enough positive feedback from contributors!

**Review by foolip — APPROVED** [review]
